<!-- KITcar launch script for preprocessing as a node -->
<launch>

    <!--<env if="$(arg debug_mode)" name="ROSCONSOLE_CONFIG_FILE" value="$(find perception)/rosconsole/perception.conf"/>-->


    <node name="runner" pkg="rtcf" type="rt_runner-gnulinux" output="screen">
        <rosparam param="mode">"wait_for_components"</rosparam>
        <!--<rosparam param="whitelist_ros_mapping">".*ros.*"</rosparam>-->
        <rosparam param="num_components_expected">2</rosparam>
        <rosparam param="frequency">100.0</rosparam>
    </node>

    <node name="foobar_1" pkg="rtcf" type="rt_dummy" args="rt_type=foobar is_first=true" output="screen">
        <remap from="out_Port" to="/mapped/out_Port_1"/>
        <remap from="in_Port_1" to="/mapped/to_ros_in_Port"/>

        <rosparam param="is_first">true</rosparam>
        <rosparam param="rt_type">"sum"</rosparam>
        <rosparam param="is_sync">true</rosparam>
        <rosparam param="topics_ignore_for_graph" >".*n-1.*"</rosparam>
    </node>

    <node name="foobar_2" pkg="rtcf" type="rt_dummy" args="rt_type=foobar is_first=false" output="screen">
        <remap from="out_Port" to="/mapped/out_Port_2_to_ros"/>
        <remap from="in_Port_1" to="/mapped/out_Port_1"/>

        <rosparam param="is_first">false</rosparam>
        <rosparam param="rt_type">"sum"</rosparam>
        <rosparam param="is_sync">true</rosparam>
        <rosparam param="topics_ignore_for_graph" >".*n-1.*"</rosparam>
    </node>

</launch>

